test/negatives/neg.inheritance-changes-types.3.ts

```diff
@@ -7,6 +7,7 @@ export class Something {
   }
 }
 
+// This should fail - covariant parameter types are not allowed
```

Comment:
Ah but are contravariant parameter types allowed though?

Because I feel if class methods are covariant in their return type is covariant, then they should also be covariant in their parameter types? (We might have had this discussion already 😉  but if so it might be worthwhile writing down somewhere...)

Reply:
Notably this is already somewhat supported in C# (and others) through overloads. We don't currently allow that in jsii, but absolutely could make it an allowed thing.

I'm guessing overloads are the reason there is very little benefit for C# to implement actual parameter covariance.

Reply:
I know they are not the same thing. I think in many cases, for jsii, they will result in a similar effect with some slight inconvenience to the Dotnet user.